Core/COEditingContext.h

```diff
@@ -15,6 +15,27 @@
 @class COError;
 
 /**
+ * The behavior to decide when the editing context should unload persistent 
+ * roots.
+ *
+ * TODO: If we keep it around, this should probably become a mask.
+ */
+typedef NS_ENUM(NSUInteger, COEditingContextUnloadingBehavior) {
+	/**
+	 * Persistent roots cannot be unloaded, except uncommitted persistent roots
+	 * on deletion.
+	 */
+	COEditingContextUnloadingBehaviorNever,
+	/**
+	 * Persistent roots can be unloaded on deletion.
+	 *
+     * For external deletions committed in other editing contexts, persistent
+	 * roots will unloaded in the current one.
+	 */
+	COEditingContextUnloadingBehaviorOnDeletion
```

Comment:
IMO, COEditingContextUnloadingBehaviorOnDeletion could be the only supported option. 

The design we have when a COEditingContext receives a notification that some changes were made in the store is, the changes take effect immediately (branch switch, current revision of branch switch, new persistent root, etc.), and there's no way to opt out of this. The rationale is, everything is undoable anyway, so it would just add complexity to have fine-grained control of propagating changes from store->editing context.